src/main/java/de/tum/in/www1/artemis/web/rest/ProgrammingExerciseParticipationResource.java

```diff
@@ -263,7 +264,34 @@ List<CommitInfoDTO> getCommitInfosForParticipationRepo(@PathVariable long partic
     public List<CommitInfoDTO> getCommitHistoryForParticipationRepo(@PathVariable long participationId) {
         ProgrammingExerciseStudentParticipation participation = programmingExerciseStudentParticipationRepository.findByIdElseThrow(participationId);
         participationAuthCheckService.checkCanAccessParticipationElseThrow(participation);
-        return programmingExerciseParticipationService.getCommitInfos(participation);
+        return programmingExerciseParticipationService.getCommitInfos(participation, false);
+    }
+
+    /**
+     * GET /programming-exercise/{exerciseID}/commit-history/{repositoryType} : Get the commit history of a programming exercise repository. The repository type can be TEMPLATE or
+     * SOLUTION or TESTS.
+     * Here we check is at least a teaching assistant for the exercise.
+     *
+     * @param exerciseID     the id of the exercise for which to retrieve the commit history
+     * @param repositoryType the type of the repository for which to retrieve the commit history
+     * @return A list of commitInfo DTOs with the commits information of the repository
+     */
+    @GetMapping("programming-exercise/{exerciseID}/commit-history/{repositoryType}")
+    @EnforceAtLeastTutor
+    public List<CommitInfoDTO> getCommitHistoryForTemplateSolutionOrTestRepo(@PathVariable long exerciseID, @PathVariable String repositoryType) {
+        boolean isTestRepository = repositoryType.equals("TESTS");
+        ProgrammingExerciseParticipation participation;
+        if (!repositoryType.equals("TEMPLATE") && !repositoryType.equals("SOLUTION") && !repositoryType.equals("TESTS")) {
+            throw new BadRequestAlertException("Invalid repository type", ENTITY_NAME, "invalidRepositoryType");
+        }
```

Comment:
Can you convert the String to an repositoryType enum instead of doing these checks manually?